src/main/java/org/testeditor/web/xtext/index/persistence/GitDiffXtextIndexTranslator.xtend

```diff
@@ -18,18 +18,23 @@ class GitDiffXtextIndexTranslator {
 			gitDiffs.forEach [
 				switch (changeType) {
 					case ADD: {
+						logger.trace("Adding file='{}' to index based on changeType='{}'.", newPath, changeType.name)
 						index.add(URI.createFileURI(newPath))
 					}
 					case COPY: {
+						logger.trace("Adding file='{}' to index based on changeType='{}'.", newPath, changeType.name)
 						index.add(URI.createFileURI(newPath))
 					}
 					case DELETE: {
+						logger.trace("Removing file='{}' from index based on changeType='{}'.", oldPath, changeType.name)
 						index.remove(URI.createFileURI(oldPath))
 					}
 					case MODIFY: {
+						logger.trace("Updating file='{}' within index based on changeType='{}'.", oldPath, changeType.name)
 						index.update(URI.createFileURI(oldPath))
 					}
 					case RENAME: {
+						logger.trace("Removing file='{}' index, adding same file='{}' based on changeType='{}'.", oldPath, newPath, changeType.name)
 						index.remove(URI.createFileURI(oldPath))
 						index.add(URI.createFileURI(newPath))
 					}
```

Comment:
Maybe extract logger-prints? Some lines are copy + pasted

Reply:
they differ a bit (usually), only one line is identical, but extraction seems to make things looking rather complex then simpler